pennylane/measurements/sample.py

```diff
@@ -172,30 +172,28 @@ def shape(self, device=None):
                     "a device with a shot vector is not supported."
                 )
             return tuple(
-                (shot_val,) if shot_val != 1 else tuple() for shot_val in device._raw_shot_sequence
+                (shot_val,) if shot_val != 1 else tuple() for shot_val in config._raw_shot_sequence
```

Comment:
We can rewrite this line to depend on `config.shot_vector` instead of the private attribute. It may no longer fit on one line, but it will make our lives easier to not have to deal with private attributes.

Reply:
if we remove the usage here, should we also not store it in the ExecutionConfig (that is, should we remove it from our informal "shots API")?

Reply:
Realistically `_raw_shot_sequence` could be removed from `ExecutionConfig` without any repercussions. But, I think if we decide to make that change it should go in a different PR so that the changes in this PR are limited to necessary changes pertaining to `MeasurementProcess.shape`, which for now means replicating shots as they are in `Device`.